### PR TITLE
Setup LLVM LIT for unit and integration tests

### PR DESCRIPTION
**Pull request by luc-tielen**
- [x] end-to-end test for compiling code in README
- [x] remove wildcards transform
- [x] unique vars transform
- [x] shift assigns transform 
- [x] AST lowering tests
- [x] RA lowering tests
- [x] EIR lowering tests
- [x] Allow no colors to be output via CLI
- [x] Typesystem (with proper error message)
- [x] Semantic analysis (with proper error message)
- [x] Parser (with proper error message)
- [x] Remove old fixture files (where possible)
- [x] Fix test failures
- [x] Make an issue for remaining tests to be added

(The other tests can stay in Haskell / hspec for now, they are at the unit test level mostly anyway.)

### CODE REVIEW

**Comment by smunix** [comment]
@luc-tielen - to run the `lit` tests, let's merge https://github.com/luc-tielen/eclair-lang/pull/65 first

File: flake.nix [hunk 1]
```diff
@@ -18,117 +18,122 @@
...
+                      doStaticLibraries = enableStaticLibraries;
+                      doSharedExecutables = disableSharedExecutables;
+                      overrideAttrs = old: {
+                        # /build/source/... is a virtual filesystem Nix uses
+                        # when building Eclair, so by adding it to the path we
+                        # can access the eclair executable during the tests.
+                        checkPhase = ''
+                          runHook preCheck
+                          DATALOG_DIR="${self}/cbits/" SOUFFLE_BIN="${final.souffle}/bin/souffle" ./Setup test
+                          PATH="${final.which}/bin:${final.souffle}/bin:/build/source/dist/build/eclair/:${llvmPackages.clang}/bin/:$PATH" ${final.lit}/bin/lit tests -v
```
Comment (kpadmasola) [inline flake.nix:91]: @luc-tielen I had to change this line to the following to get the `eclair` and `node` binaries to be found (thanks to @smunix for the tip :-)):
```bash
PATH="${final.which}/bin:${final.souffle}/bin:$PWD/dist/build/eclair:${llvmPackages.clang}/bin/:${pkgs.nodejs}/bin:$PATH" ${final.lit}/bin/lit tests -v
```

File: tests/lowering/multiple_rule_clauses.eclair [hunk 2]
```diff
@@ -0,0 +1,231 @@
+// RUN: split-file %s %t
+
+// RUN: %eclair compile --emit ra %t/program.eclair > %t/actual_ra.out
+// RUN: diff %t/expected_ra.out %t/actual_ra.out
+
+// RUN: %eclair compile --emit eir %t/program.eclair > %t/actual_eir.out
+// RUN: diff %t/expected_eir.out %t/actual_eir.out
+
```
Comment (kpadmasola) [inline tests/lowering/multiple_rule_clauses.eclair:8]: @luc-tielen We need to add the following line, which seems to have been missed here:
```bash
// RUN: %eclair compile --emit llvm %t/program.eclair > %t/actual_llvm.out
```
Once this line is added, the integration tests all pass !

Comment (luc-tielen) [reply tests/lowering/multiple_rule_clauses.eclair:8]: Oops, forgot this line! I must've forgotten tis somehow :thinking:.. thanks!

File: flake.nix (hunk 1 above)
Comment (luc-tielen) [reply flake.nix]: Thanks, this helped. Also had to explicitly add 'node' when calling the `extract_snippet` script to make it work in `nix build`

**Pull request merged**